### PR TITLE
Make constant defined checks more explicit for Rails::VERSION

### PR DESCRIPTION
**Pull request by nilsding**
This PR is made on top of #269, however that one changed the `defined?` check in just one place.  I have adapted it everywhere where it seemed to make sense.

### CODE REVIEW

**Comment by nilsding** [comment]
I have updated the `newrelic_rpm` gem to my fork for one of our services, and it now reports all errors again as expected.  Before those changes nothing was reported at all.

File: lib/new_relic/agent/instrumentation/rails3/action_controller.rb [hunk 1]
```diff
@@ -88,7 +88,7 @@ class ActionController::Base
   @name = :rails30_view
 
   depends_on do
-    defined?(::Rails) && ::Rails::VERSION::MAJOR.to_i == 3 && ::Rails::VERSION::MINOR.to_i == 0
+    defined?(::Rails::VERSION) && ::Rails::VERSION::MAJOR.to_i == 3 && ::Rails::VERSION::MINOR.to_i == 0
```
Comment (ThomasKoppensteiner) [inline lib/new_relic/agent/instrumentation/rails3/action_controller.rb]: Should it be `defined?(::Rails::VERSION::MAJOR) && defined?(::Rails::VERSION::MINOR) && ...` ?

File: lib/new_relic/agent/instrumentation/rails3/action_controller.rb [hunk 2]
```diff
@@ -127,7 +127,7 @@ def render_with_newrelic(*args, &block)
   # We can't be sure that this will work with future versions of Rails 3.
   # Currently enabled for Rails 3.1 and 3.2
   depends_on do
-    defined?(::Rails) && ::Rails::VERSION::MAJOR.to_i == 3 && ([1,2].member?(::Rails::VERSION::MINOR.to_i))
+    defined?(::Rails::VERSION) && ::Rails::VERSION::MAJOR.to_i == 3 && ([1,2].member?(::Rails::VERSION::MINOR.to_i))
```
Comment (ThomasKoppensteiner) [inline lib/new_relic/agent/instrumentation/rails3/action_controller.rb]: Should it be defined?(::Rails::VERSION::MAJOR) && defined?(::Rails::VERSION::MINOR) && ... ?

Comment (TheWudu) [reply lib/new_relic/agent/instrumentation/rails3/action_controller.rb]: That would be even safer, but not necessary imho.

**Comment by danascheider** [comment]
Thanks for the contributions @nilsding and @AlexRiedler! We appreciate the work you both put into this, and helping us make the Ruby Agent better for everyone. We’ve accepted this PR, and it will go into our next release, version 4.7.

**Pull request closed without merging**